src/main/java/soolutions/hometaskapp/account/Account.java

```diff
@@ -0,0 +1,10 @@
+package soolutions.hometaskapp.account;
+
+import soolutions.hometaskapp.core.Amount;
+
+public interface Account {
+    boolean open();
+    void deposit(Amount amount);
```

Comment:
Does it make sense to return the current balance here?

Reply:
No! I am following command/query technique here.

Reply:
Agreed. Read and writes have very different purposes. Command/query separation seems good.

Reply:
@SilasReinagel if I go for command/query separation the immutability is difficult because I then need to return the new object.

Reply:
Correct. Which might be why @goqp was proposing that Objects might return an instance of themselves when impacted by mutators. But, that can be designed in a future ticket. We don't need it yet.